core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -487,14 +519,21 @@ parallel_reduce(const Impl::TeamThreadRangeBoundariesStruct<
                     iType, Impl::CudaTeamMember>& loop_boundaries,
                 const Closure& closure, const ReducerType& reducer) {
   KOKKOS_IF_ON_DEVICE(
-      (typename ReducerType::value_type value;
+      (using value_type            = typename ReducerType::value_type;
+       using functor_analysis_type = typename Impl::FunctorAnalysis<
+           Impl::FunctorPatternInterface::REDUCE,
+           TeamPolicy<typename Impl::CudaTeamMember::execution_space>,
+           ReducerType, value_type>;
+       using wrapped_reducer_type = typename functor_analysis_type::Reducer;
 
-       reducer.init(value);
+       wrapped_reducer_type wrapped_reducer(reducer); value_type value{};
```

Comment:
We don't really need the explicit initialization here. If we omit it, there is a higher chance of catching bugs in the implementation.
```suggestion
       wrapped_reducer_type wrapped_reducer(reducer); value_type value;
```